migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/HttpSettingsToHttpClient.java

```diff
@@ -15,57 +15,444 @@
-                         .filter(t -> t.isAssignableTo(ClientOverrideConfiguration.class.getCanonicalName()))
-                         .isPresent()) {
+                return saveHttpSettingsToHttpClientBuilderCursorIfNeeded(method, configToHttpSettings, methodInvocationName);
+
+            }
+
+            return method;
+        }
+
+        private J.MethodInvocation saveHttpSettingsToHttpClientBuilderCursorIfNeeded(
+            J.MethodInvocation method,
+            Map<String, Map<String, Expression>> configToHttpSettings,
+            String configKey) {
+
+            if (CollectionUtils.isNullOrEmpty(configToHttpSettings) ||
+                !configToHttpSettings.containsKey(configKey)) {
                 return method;
             }
 
+            Cursor parentCursor = getCursor().dropParentUntil(
+                parent -> (parent instanceof J.MethodInvocation) && ((J.MethodInvocation) parent)
+                    .getSimpleName().equals("build"));
+
+            parentCursor.putMessage(HTTP_CLIENT_BUILDER_HTTP_SETTINGS_CURSOR, configToHttpSettings.get(configKey));
+            return method;
+        }
+
+        private Pair<Class, Class> httpClientClassNamePair(JavaType.FullyQualified classType) {
+            Pair<Class, Class> httpClientClassNamePair;
+            if (!classType.getClassName().contains("Async")) {
+                httpClientClassNamePair = Pair.of(ApacheHttpClient.class,
+                                                  ApacheHttpClient.Builder.class);
+            } else {
+                httpClientClassNamePair = Pair.of(NettyNioAsyncHttpClient.class,
+                                                  NettyNioAsyncHttpClient.Builder.class);
+            }
+            return httpClientClassNamePair;
+        }
+
+        private JContainer<Expression> httpClientBuilderInvoke(Map<String, Expression> httpSettings,
+                                                               Pair<Class, Class> httpClientClassNamePair) {
+            Class httpClientClassName = httpClientClassNamePair.left();
+            Class httpClientBuilderClassName = httpClientClassNamePair.right();
+
+            JavaType.FullyQualified httpClientType =
+                TypeUtils.asFullyQualified(JavaType.buildType(httpClientClassName.getCanonicalName()));
+            JavaType.FullyQualified httpClientBuilderType =
+                TypeUtils.asFullyQualified(JavaType.buildType(httpClientBuilderClassName.getCanonicalName()));
+
+
+            JavaType.Method httpClientBuilder = new JavaType.Method(
+                null,
+                0L,
+                httpClientType,
+                "builder",
+                httpClientBuilderType,
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyList()
+            );
+
+            J.Identifier httpClient = new J.Identifier(
```

Comment:
Might be able to use `IdentifierUtils` to remove some boilerplate for creating these

Reply:
Ah nice, I was not aware of it